### PR TITLE
Feat add deletes worker

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

Move function deletion operations from the functions controller to the deletes worker. 

## Test Plan

Use Existing tests

## Related PRs and Issues
NA

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
YES

### CODE REVIEW

File: app/executor.php [hunk 1]
```diff
@@ -748,6 +746,7 @@ function runBuildStage(string $buildId, string $deploymentId, string $projectID)
         }
 
         $vars = $resource->getAttribute('vars', []);
+        $vars['ENTRYPOINT_NAME'] = $resource->getAttribute('entrypoint', '');
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/executor.php:749]: With the new executor refactor, this environment variable is no longer required.
https://github.com/appwrite/appwrite/blob/9262bad8bf8fd37947e06b9883d6473f1ecb4244/app/executor.php#L284

File: app/executor.php [hunk 2]
```diff
@@ -1045,40 +1046,43 @@ function (string $functionId, Response $response, Database $dbForProject) use ($
...
+                $redis = $register->get('redisPool')->get();
+                $cache = new Cache(new RedisCache($redis));
+                $dbForProject = new Database(new MariaDB($db), $cache);
+                $dbForProject->setDefaultDatabase(App::getEnv('_APP_DB_SCHEMA', 'appwrite'));
+                $dbForProject->setNamespace('_project_' . $projectId);
+
+                $builds = $dbForProject->find('builds', [ 
+                    new Query('deploymentId', Query::TYPE_EQUAL, [$deploymentId]),
+                    new Query('status', Query::TYPE_EQUAL, ['building'])
+                ], 999);
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/executor.php:1071]: We're completely removing the database dependencies from the executor so this code will be removed 👍

File: app/workers/deletes.php [hunk 3]
```diff
@@ -307,24 +311,139 @@ protected function deleteAuditLogsByResource(string $resource, string $projectId
...
+        try {
+            $ch = \curl_init();
+            \curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
+            \curl_setopt($ch, CURLOPT_URL, "http://appwrite-executor/v1/functions/{$document->getId()}");
+            \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+                'Content-Type: application/json',
+                'x-appwrite-project: '. $projectId,
+                'x-appwrite-executor-key: '. App::getEnv('_APP_EXECUTOR_SECRET', '')
+            ]);
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/workers/deletes.php:327]: @eldadfux suggested using an `Executor.php` class which has been added in the new PR 

https://github.com/appwrite/appwrite/blob/9262bad8bf8fd37947e06b9883d6473f1ecb4244/src/Executor/Executor.php

https://github.com/appwrite/appwrite/pull/2739

File: app/workers/deletes.php [hunk 4]
```diff
@@ -369,9 +488,7 @@ protected function deleteForProjectIds(callable $callback): void
         $executionStart = \microtime(true);
 
         while ($sum === $limit) {
-            Authorization::disable();
-            $projects = $this->getConsoleDB()->find('projects', [], $limit, ($chunk * $limit));
-            Authorization::reset();
+            $projects = Authorization::skip(fn() => $this->getConsoleDB()->find('projects', [], $limit, ($chunk * $limit)));
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/workers/deletes.php:491]: Done

File: app/controllers/api/functions.php [hunk 5]
```diff
@@ -368,6 +368,7 @@
 
         $function = $dbForProject->getDocument('functions', $functionId);
         $deployment = $dbForProject->getDocument('deployments', $deployment);
+        var_dump($deployment);
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/controllers/api/functions.php]: removed.

File: app/workers/deletes.php [hunk 6]
```diff
@@ -307,24 +313,139 @@ protected function deleteAuditLogsByResource(string $resource, string $projectId
...
+
+            \curl_close($ch);
+        } catch (Throwable $th) {
+            Console::error($th->getMessage());
+        }
+
+        /**
+         * Delete deployment files
+         */
+        $storageFunctions = new Local(APP_STORAGE_FUNCTIONS . '/app-' . $projectId);
```
Comment (lohanidamodar) [inline app/workers/deletes.php:428]: Remember use of storage has to be refactored everywhere, once merged with the storage bucket changes.

**Review by lohanidamodar — COMMENTED** [review]
Looks good, all the concerns seems to be addressed in executor refactor, will review that next.

File: app/workers/deletes.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (eldadfux) [reply app/workers/deletes.php:329]: Because we're decoupling this outside of the repo, no reason to tackle this issue now.

**Review by eldadfux — APPROVED** [review]

**Review by lohanidamodar — APPROVED** [review]

**Comment by christyjacob4** [comment]
Thank you @lohanidamodar and @eldadfux

**Pull request merged**